selectinf/randomized/query.py

```diff
@@ -1669,22 +1405,36 @@ def selective_MLE(observed_target,
     observed_target = np.atleast_1d(observed_target)
     prec_target = np.linalg.inv(target_cov)
 
+    prec_opt = np.linalg.inv(cond_cov)
+
     # target_lin determines how the conditional mean of optimization variables
     # vary with target
     # logdens_linear determines how the argument of the optimization density
     # depends on the score, not how the mean depends on score, hence the minus sign
 
-    target_lin = - logdens_linear.dot(target_score_cov.T.dot(prec_target)) 
-    target_offset = cond_mean - target_lin.dot(observed_target)
+    target_linear = target_score_cov.T.dot(prec_target)
+    target_offset = score_offset - target_linear.dot(observed_target)
 
-    prec_opt = np.linalg.inv(cond_cov)
+    target_lin = - logdens_linear.dot(target_linear)
+    target_off = cond_mean - target_lin.dot(observed_target)
+
+    if np.asarray(randomizer_prec).shape in [(), (0,)]:
+        _P = target_linear.T.dot(target_offset) * randomizer_prec
+        _prec = prec_target + (target_linear.T.dot(target_linear) * randomizer_prec) - target_lin.T.dot(prec_opt).dot(
+            target_lin)
+    else:
+        _P = target_linear.T.dot(randomizer_prec).dot(target_offset)
```

Comment:
It seems that it is only here that `randomizer_prec` is used -- this is why split_lasso is sort of broken now of course. The actual matrix product we need is `target_score_cov.dot(prec_target)`. In general we won't be able to do this unless we knew the target. Hmm... if we knew the target were of the form (or its projection onto score) `A.dot(score)` and we wrote things in terms of A everywhere, then we would only need to know the product `A.dot(randomizer_prec)`.